<launch>
  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

  <node pkg="nodelet" type="nodelet" name="rgb_depth_register_node"
        args="load depth_image_proc/register nodelet_manager">
    <remap from="rgb/camera_info" to="/thorvald_001/kinect2_right_camera/hd/camera_info"/>
    <remap from="depth/camera_info" to="/thorvald_001/kinect2_right_sensor/sd/camera_info"/>
    <remap from="depth/image_rect" to="/thorvald_001/kinect2_right_sensor/sd/image_depth_rect"/>
  </node>
  
  <node pkg="nodelet" type="nodelet" name="rgb_pointcloud_node"
        args="load depth_image_proc/point_cloud_xyzrgb nodelet_manager">
    <remap from="rgb/camera_info" to="/thorvald_001/kinect2_right_camera/hd/camera_info"/>
    <remap from="rgb/image_rect_color" to="/thorvald_001/kinect2_right_camera/hd/image_color_rect"/>
    <remap from="depth/image_rect" to="/depth_registered/image_rect"/>
  </node>

</launch>
